theano/compile/builders.py

```diff
@@ -141,6 +141,11 @@ def connection_pattern(self, node):
         """
         return io_connection_pattern(self.new_inputs, self.new_outputs)
 
+    def infer_shape(self, node, shapes):
+        return theano.scan_module.scan_utils.infer_shape(self.new_outputs,
+                                                         self.new_inputs,
+                                                         shapes)
```

Comment:
Using scan_utils.infer_shape returns symbolic shapes potentially computed using the inputs of the inner function but we need them to be computed using the inputs of the OpFromGraph in the outer function. This means that you need to do a clone (with replace) on the output of scan_utils.infer_shape to replace the inner inputs by the outer inputs.